.github/actions/collect-e2e-artifacts/action.yaml

```diff
@@ -0,0 +1,81 @@
+name: "Collects and uploads E2E artifacts"
```

Comment:
Why move this now? I think it could use a separate commit at least.

Reply:
It's reused in two actions running e2e, extracting it avoids code duplication.

Reply:
extracted into separate commit